src/interface.jl

```diff
@@ -83,33 +83,67 @@ Identity(::LieGroup{𝔽,O}) where {𝔽,O<:AbstractGroupOperation} = Identity{O
 Identity(::O) where {O<:AbstractGroupOperation} = Identity(O)
 Identity(::Type{O}) where {O<:AbstractGroupOperation} = Identity{O}()
 
+"""
+    AbstractLieGroupPoint <: ManifoldsBase.AbstractManifoldPoint end
+
+An abstract type for a point on a [`LieGroup`](@ref).
+While an points and tangent vectors are usually kept untyped for flexibility,
+it might be necessary to distinguish different types of points, for example
+
+* for complicated representations that require a struct
+@ semantic verification
+* when there exist different representations
+
+By sub-typing the [`AbstractManifoldPoint`](@extref `ManifoldsBase.AbstractManifoldPoint`), this follows the same idea as in $(_link(:ManifoldsBase)).
+"""
+abstract type AbstractLieGroupPoint <: ManifoldsBase.AbstractManifoldPoint end
+
+"""
+    AbstractLieAlgebraTangentVector <: ManifoldsBase.AbstractTangentVector
+
+An abstract type for a tangent vector represented in a [`LieAlgebra`](@ref).
+
+While an tangent vectors are usually kept untyped for flexibility,
+it might be necessary to distinguish different types of points, for example
+
+* for complicated representations that require a struct
+@ semantic verification
+* when there exist different representations
+
+By sub-typing the [`AbstractManifoldPoint`](@extref `ManifoldsBase.AbstractManifoldPoint`),
+this follows the same idea as in $(_link(:ManifoldsBase)).
+"""
+abstract type AbstractLieAlgebraTangentVector <: ManifoldsBase.AbstractTangentVector end
+
 #
 #
 # --- Functions ---
 
 # Internal pass through for coordinates and vectors
 
 @inline function ManifoldsBase._get_coordinates(
-    G::LieGroup, p, X, B::LieAlgebraOrthogonalBasis
+    G::LieGroup, X, B::DefaultLieAlgebraOrthogonalBasis
 )
-    return get_coordinates_lie(G, p, X, number_system(B))
+    return get_coordinates_lie(G, X, B)
 end
 @inline function ManifoldsBase._get_coordinates!(
-    G::LieGroup, Y, p, X, B::LieAlgebraOrthogonalBasis
+    G::LieGroup, c, X, B::DefaultLieAlgebraOrthogonalBasis
 )
-    return get_coordinates_lie!(G, Y, p, X, number_system(B))
+    return get_coordinates_lie!(G, c, X, B)
 end
-@inline function ManifoldsBase._get_vector(G::LieGroup, p, c, B::LieAlgebraOrthogonalBasis)
-    return get_vector_lie(G, p, c, number_system(B))
+@inline function ManifoldsBase._get_vector(
```

Comment:
I think we don't have any tests for `get_vector` with `LieGroup` as the first argument here. I'm not sure if we need it when it works for `LieAlgebra` without the point already?